### ISSUE TITLE
Graph visualize

### ISSUE THREAD

**Issue by Alek-dr** [chiphuyen/tf-stanford-tutorials#16]
Hello, please help
then i run 07_convnet_mnist.py graph looks good
![png](https://cloud.githubusercontent.com/assets/11842409/25534371/3da2b35c-2c3c-11e7-8abd-f410a3127a20.png)

but then i run  very similar code with my data (code below), graph looks awfully
and may be it absolutely another problem but accuracy not increase
![png 1](https://cloud.githubusercontent.com/assets/11842409/25534550/c356a292-2c3c-11e7-8f9f-e6aef2ccfe7a.png)

```
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from data3 import DataSet

import os
import time

import tensorflow as tf

width = 28
height = 28

ds = DataSet(batch=128, w=width, h=height)
ds.train_directory = '/home/alexander/PycharmProjects/Diplom/DB/train'
ds.test_directory = '/home/alexander/PycharmProjects/Diplom/DB/test'

N_CLASSES = 4

LEARNING_RATE = 0.001
BATCH_SIZE = 128
SKIP_STEP = 10
DROPOUT = 0.75
N_EPOCHS = 1

with tf.name_scope('data'):
    X = tf.placeholder(tf.float32, [None, 784], name="X_placeholder")
    Y = tf.placeholder(tf.float32, [None, N_CLASSES], name="Y_placeholder")

global_step = tf.Variable(0, dtype=tf.int32, trainable=False, name='global_step')

with tf.variable_scope('conv1') as scope:
    images = tf.reshape(X, shape=[-1, width, height, 1])
    kernel = tf.get_variable('kernel', [5, 5, 1, 32],
                             initializer=tf.truncated_normal_initializer())
    biases = tf.get_variable('biases', [32],
                             initializer=tf.random_normal_initializer())
    conv = tf.nn.conv2d(images, kernel, strides=[1, 1, 1, 1], padding='SAME')
    conv1 = tf.nn.relu(conv + biases, name=scope.name)

with tf.variable_scope('pool1') as scope:
    pool1 = tf.nn.max_pool(conv1, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1],
                           padding='SAME')

with tf.variable_scope('conv2') as scope:
    kernel = tf.get_variable('kernels', [5, 5, 32, 64],
                             initializer=tf.truncated_normal_initializer())
    biases = tf.get_variable('biases', [64],
                             initializer=tf.random_normal_initializer())
    conv = tf.nn.conv2d(pool1, kernel, strides=[1, 1, 1, 1], padding='SAME')
    conv2 = tf.nn.relu(conv + biases, name=scope.name)

with tf.variable_scope('pool2') as scope:
    pool2 = tf.nn.max_pool(conv2, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1],
                           padding='SAME')

with tf.variable_scope('fc') as scope:
    input_features = 7 * 7 * 64
    w = tf.get_variable('weights', [input_features, 1024],
                        initializer=tf.truncated_normal_initializer())
    b = tf.get_variable('biases', [1024],
                        initializer=tf.random_normal_initializer())
    pool2 = tf.reshape(pool2, [-1, input_features])
    fc = tf.nn.relu(tf.matmul(pool2, w) + b, name='relu')

    fc = tf.nn.dropout(fc, 0.75, name='relu_dropout')

with tf.variable_scope('softmax_linear') as scope:
    w = tf.get_variable('weights', [1024, N_CLASSES],
                        initializer=tf.truncated_normal_initializer())
    b = tf.get_variable('biases', [N_CLASSES],
                        initializer=tf.random_normal_initializer())
    logits = tf.matmul(fc, w) + b

with tf.name_scope('loss'):
    entropy = tf.nn.softmax_cross_entropy_with_logits(logits=logits, labels=Y)
    loss = tf.reduce_mean(entropy, name='loss')

optimizer = tf.train.AdamOptimizer(LEARNING_RATE).minimize(loss,
                                                           global_step=global_step)

with tf.Session() as sess:
    sess.run(tf.global_variables_initializer())
    saver = tf.train.Saver()
    writer = tf.summary.FileWriter('tb', sess.graph)
    ckpt = tf.train.get_checkpoint_state(os.path.dirname('tb/checkpoint'))
    if ckpt and ckpt.model_checkpoint_path:
        saver.restore(sess, ckpt.model_checkpoint_path)

    initial_step = global_step.eval()

    start_time = time.time()
    n_batches = int(ds.num_train_files / BATCH_SIZE)

    total_loss = 0.0
    total_correct_preds = 0
    for index in range(initial_step, n_batches * N_EPOCHS):
        X_batch, Y_batch = ds.next_train_batch()
        _, loss_batch, logits_batch = sess.run([optimizer, loss, logits],
                                               feed_dict={X: X_batch, Y: Y_batch})
        preds = tf.nn.softmax(logits_batch)
        correct_preds = tf.equal(tf.argmax(preds, 1), tf.argmax(Y_batch, 1))
        accuracy = tf.reduce_sum(tf.cast(correct_preds, tf.float32))
        total_correct_preds += sess.run(accuracy)
        total_loss += loss_batch
        if (index + 1) % SKIP_STEP == 0:
            print('Average loss at step {}: {:5.1f}'.format(index + 1, total_loss / SKIP_STEP))
            print('Accuracy: ', total_correct_preds / (BATCH_SIZE * 10))
            total_loss = 0.0
            total_correct_preds = 0
            saver.save(sess, 'tb', index)

    print("Optimization Finished!")
    print("Total time: {0} seconds".format(time.time() - start_time))

    n_batches = int(ds.num_test_files / BATCH_SIZE)
    total_correct_preds = 0
    for i in range(n_batches):
        X_batch, Y_batch = ds.next_test_batch()
        _, loss_batch, logits_batch = sess.run([optimizer, loss, logits],
                                               feed_dict={X: X_batch, Y: Y_batch})
        preds = tf.nn.softmax(logits_batch)
        correct_preds = tf.equal(tf.argmax(preds, 1), tf.argmax(Y_batch, 1))
        accuracy = tf.reduce_sum(tf.cast(correct_preds, tf.float32))
        total_correct_preds += sess.run(accuracy)

    print("Accuracy {0}".format(total_correct_preds / ds.num_test_files))
```

**Comment by chiphuyen** [comment]
@Alek-dr You should put all code processing data into a scope, because the processing code is what crowds your graph. Sorry for the late response.
